### ISSUE TITLE
Words are sometimes omitted between pages in a TextBox.

### ISSUE THREAD

**Issue by danadrew** [moai/moai-dev#168]
See sample: samples / test / textbox-paging

**Expected:**
All words in the inputted text should be displayed even if multiple styles are used.

**Actual:**
If multiple different styles are used, sometimes words will be lost between pages.

**Reproduction Steps:**
1. Create a textbox that uses several different styles and spans for multiple pages.
2. When going to the next page notice that sometimes words are lost.

**Additional Info:**
I couldn't narrow down the cause that far, but I did get a repro case where it happens every time. It seems to happen pretty frequently when using multiple styles, and when it does happen it will always behave that way every time you run it. It would generally omit anywhere from one to four words. In my sample, four words are omitted.

**Comment by danadrew** [comment]
I added another sample of an issue that may be related to this one. It could also be a different issue, so I can file a separate bug if that's the case. The sample is in: samples / test / textbox-paging2

I found a case where the last word on the first page is repeated on the second page. In this sample I modified the default style, but stuck to using just that one style. It appears to have to do with a new line in the text where the page break falls.
